File: MediCareWebSpa/src/app/DTOs/request/user-register-request.dto.ts
import { RoleType } from '../../enums/role-type';
import { RegisterRequestDTO } from './register-request.dto';

export class UserRegisterRequestDTO extends RegisterRequestDTO {
  employmentDate: Date | null;
  specialityId: number | null;
  birthDate: Date | null;

  constructor(
    name: string,
    surname: string,
    email: string,
    password: string,
    pesel: string,
    phoneNumber: string,
    roleType: RoleType,
    employmentDate: Date,
    specialityId: number,
    birthDate: Date
  ) {
    super(name, surname, email, password, pesel, phoneNumber, roleType);
    this.employmentDate = employmentDate;
    this.specialityId = specialityId;
    this.birthDate = birthDate;
  }
}

File: MediCareWebApi/DTOs/Request/UserRegisterRequestDTO.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.DTOs.Request
{
    public class UserRegisterRequestDTO : RegisterRequestDTO
    {
		[Required(ErrorMessage = "Typ roli jest wymagany.")]
		public RoleType RoleType { get; set; }

		[DataType(DataType.Date, ErrorMessage = "Nieprawidłowy format daty.")]
		public DateTime? EmploymentDate { get; set; }

		public int? SpecialityId { get; set; }

		[DataType(DataType.Date, ErrorMessage = "Nieprawidłowy format daty.")]
		public DateTime? BirthDate { get; set; }
	}
}
